zerver/views/push_notifications.py

```diff
@@ -210,3 +216,12 @@ def self_hosting_auth_json_endpoint(
     redirect_url = self_hosting_auth_view_common(request, user_profile, next_page)
 
     return json_success(request, data={"billing_access_url": redirect_url})
+
+
+def self_hosting_auth_not_configured(request: HttpRequest) -> HttpResponse:
+    if settings.CORPORATE_ENABLED or uses_notification_bouncer():  # nocoverage
+        # This error page should only be available if the config error
+        # is actually real.
+        return render(request, "404.html", status=404)
```

Comment:
A question that's bugging me here is whether this endpoint should be blocked for users who aren't billing admins (to match the access permission for the regular /self-hosted-billing/ endpoints? The abstract concern would be leaking configuration info, but I don't know, `CORPORATE_ENABLED` or whether the bouncer setting is set are not exactly secret information.

Reply:
Good thing to think about. Yeah I don't think either of those are items we wouldn't happily include in `/api/v1/server-settings`; they're certainly effectively present in `/register` data available to every logged-in user.

We require authentication for the endpoint via decorators, so I think it's a non-issue.

Reply:
Hmm yeah, though this is a new `/self-hosted-billing/not-configured/` endpoint (added as a separate endpoint to allow nice handling by the desktop app) and not registered as a `rest_path`, but vanilla Django `path`, so without the benefits of our decorators.

So based on that I figured it's better to protect this same as the main endpoints, and pushed an extra commit for this.